src/components/board.tsx

```diff
@@ -13,19 +13,42 @@ type Props = {
 export const Board = (props: Props) => {
   const squaresMap = useContext(SquareMapContext);
   const [isClickEnabled, setClickEnabled] = useState(true);
+  const [change, setChange] = useState(0);
   const [getFirstSelectedSquare, setFirstSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+    useState<NullableSelectedSquare>(null);
 
   const [getSecondSelectedSquare, setSecondSelectedSquare] =
-    useState<SelectedSquare | null>(null);
+      useState<NullableSelectedSquare>(null);
 
+  useEffect(() => {
+      const values = Array.from(squaresMap.values());
+      const isCompleted = values
+          .map((k) => k.getClassName === "matched")
+          .every((k) => k);
```

Comment:
Why not directly:
`values.every((square) => square.getClassName === "matched")`?

Reply:
done